Mosaic view

bnmajor opened the pull request.

manthey commented:
How does one remove a filter?

manthey commented:
When changing the preview size to a small size, the older, larger previews appear so that they extend outside of the individual chip space.  Should we scale images to fit instead?

manthey commented:
I realize that the actions aren't yet hooked up.  When I select the "Change Class" dropdown, it would be nice if it scrolled the dropdown into view (or we left empty space below so one could scroll up as needed).  Once one selects a class in the change class dropdown, would this happen immediately?  In which case if you are filtering by class would these chips vanish?  Or would we have a confirm button?

manthey commented:
We may want some ability to hide the labelled fill color.  When reviewing, the color overlay can make it harder to verify the image.

manthey commented:
Aside from noting having a way to clear filters (unless I missed it), this looks very good.

All of my comments can be pushed off to future PRs.

manthey reviewed (approved):
I'd recommend merging this and addressing comments in future PRs.

bnmajor commented:
Replying to all comments for reference:

> Probably not related to this PR (but I'm currently on this branch so haven't checked), when I am in Labeling mode and am showing the predictions, I can't change the label of a single superpixel. It seems to ignore clicking on the superpixel, but it does still support drag-painting.

Opened as a new issue in #129

> How does one remove a filter?

The tooltip above the drop-down menu notes that ctrl + click will de-select the last filter. The tooltip is not very obvious though, so for simplicity a "clear all" button has been added in #130.

> When changing the preview size to a small size, the older, larger previews appear so that they extend outside of the individual chip space. Should we scale images to fit instead?

Yes! Also added in #130.

> I realize that the actions aren't yet hooked up. When I select the "Change Class" dropdown, it would be nice if it scrolled the dropdown into view (or we left empty space below so one could scroll up as needed).

Makes sense! This will be fixed in #130 as well.

> Once one selects a class in the change class dropdown, would this happen immediately? In which case if you are filtering by class would these chips vanish? Or would we have a confirm button?

Updated #131 with this comment so that discussion/changes are associated with the branch that will address this.

> We may want some ability to hide the labelled fill color. When reviewing, the color overlay can make it harder to verify the image.

Sounds good! Opened as a new issue in #132
